src/neo4j_graphrag/llm/mistralai_llm.py

```diff
@@ -0,0 +1,118 @@
+#  Copyright (c) "Neo4j"
+#  Neo4j Sweden AB [https://neo4j.com]
+#  #
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  #
+#      https://www.apache.org/licenses/LICENSE-2.0
+#  #
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+from __future__ import annotations
+
+import os
+from typing import Any, Optional, Union
+
+from ..exceptions import LLMGenerationError
+from .base import LLMInterface
+from .types import LLMResponse
+
+try:
+    from mistralai import Mistral
+    from mistralai.models.assistantmessage import AssistantMessage
+    from mistralai.models.sdkerror import SDKError
+    from mistralai.models.systemmessage import SystemMessage
+    from mistralai.models.toolmessage import ToolMessage
+    from mistralai.models.usermessage import UserMessage
+
+    MessageType = Union[AssistantMessage, SystemMessage, ToolMessage, UserMessage]
+except ImportError:
+    Mistral = None  # type: ignore
+    SDKError = None  # type: ignore
+
+
+class MistralAILLM(LLMInterface):
+    def __init__(
+        self,
+        model_name: str,
+        model_params: Optional[dict[str, Any]] = None,
+        **kwargs: Any,
+    ):
+        """
+
+        Args:
+            model_name (str):
+            model_params (str): Parameters like temperature and such that will be
+             passed to the chat completions endpoint
+            kwargs: All other parameters will be passed to the Mistral client.
+
+        """
+        if Mistral is None:
+            raise ImportError(
+                "Could not import Mistral Python client. "
+                "Please install it with `pip install mistralai`."
+            )
+        super().__init__(model_name, model_params)
+        api_key = os.getenv("MISTRAL_API_KEY", "")
+        self.client = Mistral(api_key=api_key, **kwargs)
+
+    def get_messages(self, input: str) -> list[MessageType]:
```

Comment:
When I see the complexity for the type, I wonder if this `get_messages` method is not just over-complicating things...

Also, I guess that you tried `list[UserMessage]` and it's not compatible with the mistral signature or something?

Reply:
Yep, it raises a mypy error unfortunately. We could just do without it perhaps

Reply:
Passing `[UserMessage(content=input)]` directly doesn't work too so having `get_messages` here may be the cleanest solution

Reply:
> Passing [UserMessage(content=input)] directly doesn't work

Super weird, but let's not fight against that, if it works, I'm fine :)